relay-server/src/utils/dynamic_sampling.rs

```diff
@@ -15,79 +15,63 @@ use crate::envelope::{Envelope, ItemType};
 /// Checks whether an event should be kept or removed by dynamic sampling.
 pub fn should_keep_event(
     sampling_context: Option<&DynamicSamplingContext>,
-    event: &Event,
+    event: Option<&Event>,
     ip_addr: Option<IpAddr>,
     project_state: &ProjectState,
+    trace_root_project_state: Option<&ProjectState>,
```

Comment:
👍  for explicit naming, in principle. Please ensure that whichever terminology we want to establish is used consistently throughout the code base; most places refer to this as `sampling_project_state`. 

Develop docs also refer to the context as "Dynamic Sampling Context" (see [here](https://develop.sentry.dev/sdk/performance/dynamic-sampling-context/)), so we should best stick to that terminology and call this `sampling_project_state` as well.

Reply:
You are right about consistency, I just don't like the name. I found it odd to call it "sampling" project state here though because obviously both are being used for sampling, because a sampling-related function is being called.